tests/pytests/unit/states/saltmod/test_function.py

```diff
@@ -0,0 +1,527 @@
+    tgt = "larry"
+
+    expected = {
+        "name": name,
+        "changes": {"ret": {tgt: ""}},
+        "result": True,
+        "comment": f"Function ran successfully. Function state ran on {tgt}.",
+    }
+
+    with patch.dict(saltmod.__opts__, {"test": False}):
+        mock = MagicMock(
+            return_value={
+                tgt: {
+                    "ret": "",
+                    "retcode": 0,
+                    "failed": False,
+                },
+            },
+        )
+        batch = "yes"
+        with patch.dict(saltmod.__salt__, {"saltutil.cmd": mock}):
+            ret = saltmod.function(name, tgt, batch=batch)
+            assert ret == expected
+        mock.assert_called_once()
+        assert "batch" in mock.call_args.kwargs
+        assert mock.call_args.kwargs["batch"] == batch
+
+        batch = ["yes", "no"]
+        with patch.dict(saltmod.__salt__, {"saltutil.cmd": mock}):
+            ret = saltmod.function(name, tgt, batch=batch)
+            assert ret == expected
+        assert "batch" in mock.call_args.kwargs
+        assert mock.call_args.kwargs["batch"] == str(batch)
+
+
+def test_subset():
+    name = "state"
+    tgt = "larry"
+
+    expected = {
+        "name": name,
+        "changes": {"ret": {tgt: ""}},
+        "result": True,
+        "comment": f"Function ran successfully. Function state ran on {tgt}.",
+    }
+
+    with patch.dict(saltmod.__opts__, {"test": False}):
+        mock = MagicMock(
+            return_value={
+                tgt: {
+                    "ret": "",
+                    "retcode": 0,
+                    "failed": False,
+                },
+            },
+        )
+        subset = "yes"
+        with patch.dict(saltmod.__salt__, {"saltutil.cmd": mock}):
+            ret = saltmod.function(name, tgt, subset=subset)
+            assert ret == expected
+        mock.assert_called_once()
+        assert "subset" in mock.call_args.kwargs
+        assert mock.call_args.kwargs["subset"] == subset
+
+
+def test_ret_config():
+    name = "state"
+    tgt = "larry"
+
+    expected = {
+        "name": name,
+        "changes": {"ret": {tgt: ""}},
+        "result": True,
+        "comment": f"Function ran successfully. Function state ran on {tgt}.",
+    }
+
+    with patch.dict(saltmod.__opts__, {"test": False}):
+        mock = MagicMock(
+            return_value={
+                tgt: {
+                    "ret": "",
+                    "retcode": 0,
+                    "failed": False,
+                },
+            },
+        )
+        ret_config = {"yes": "no"}
+        with patch.dict(saltmod.__salt__, {"saltutil.cmd": mock}):
+            ret = saltmod.function(name, tgt, ret_config=ret_config)
+            assert ret == expected
+        mock.assert_called_once()
+        assert "ret_config" in mock.call_args.kwargs
+        assert mock.call_args.kwargs["ret_config"] == ret_config
+
+
+def test_ret_kwargs():
+    name = "state"
```

Comment:
Reading the docs of saltmod.function it's expecting name to be `cmd.run` or `pkg.install`, but this `name` in your test only includes the name of the module not the function. This applies to the rest of the tests that use this same `name`.

Reply:
This particular name is the name we set on sls files:
```yaml
<name>:
  salt.function:
    - name: pkg.installed
```
Or
```yaml
vim:
  salt.function:
    - name: pkg.installed
```
Or, a more complete example
```yaml
delete-root:
  state.function:
    - name: cmd.run
    - tgt: '*'
    - arg:
      - rm -rf /
```
So, it can really be the salt function name to run, or whatever arbitrary name we want.